### ISSUE TITLE
Cartesian products are not extended to SimpleWeightedGraphs

### ISSUE THREAD

**Issue by bubblingoak** [sbromberger/SimpleWeightedGraphs.jl#39]
When constructing a more complex graph that is logically represented as a cartesian product of two (or more) other graphs, it can be useful to use `cartesian_product` to simplify the code; however this function produces an illogical result for SimpleWeightedGraphs. Namely, it produces a graph with the correct structure, but with all weights set to 1.0. This seems like very non-ideal and confusing behavior. The `cartesian_product` function seems easily extendible to weighted graphs, but at the very least I'd think this should throw a warning/error.

As an example,
```
g = SimpleWeightedGraph(5);
add_edge!(g, 1, 2, 1.0);
add_edge!(g, 1, 3, 1.0);
add_edge!(g, 1, 4, 0.5);
add_edge!(g, 1, 5, 0.5);
 
h = SimpleWeightedGraph(3);
add_edge!(h, 1, 2, 9.0);
add_edge!(h, 2, 3, 1.1);
add_edge!(h, 3, 1, 2.0);

# Produces a {15,27} undirected simple Int64 graph with Float64 weights, but the weights are all 1.0
cartesian_product(g,h)
```

**Comment by sbromberger** [comment]
In general, one should not expect that graph operators from LG core will work without overrides for other graph types. That said, we have developed some operations for SWG in the past and would love to get more. PRs gratefully accepted :)

**Comment by bubblingoak** [comment]
Fair point. I suppose the real issue is that LG core packages accept the type `AbstractGraph`, while treating it as a simple graph with no weights. SWG then places itself in the hierarchy as a subtype of `AbstractGraph` when in reality it is more general than the graphs used in LG and unlike a typical subtype, not all operations which can be performed on the supertype are valid with the subtype. That said, I presume there are other reasons for doing it this way rather than through inheritance; I'm rather new to Julia.

Anyhow, I've submitted a PR with a trivial override utilizing a nearly identical implementation to the original LG function :)

i.e. added the weight input to `add_edge!`
`add_edge!(z, id(i1, j), id(i2, j))`  -> `add_edge!(z, id(i1, j), id(i2, j), weight(e))`

**Comment by sbromberger** [comment]
Closed via #40.
